### PR TITLE
EES-3835 Fix login redirect

### PR DESCRIPTION
**Pull request by markhiveit**
This PR fixes a bug where after logging in, a user wouldn't get redirected to page they were previously trying to access.

This was fixed by ensuring the argument name for the return URL was correct on `OnGetCallbackAsync`. It must be `returnUrl`, otherwise IdentityServer4 will always return null for that argument.

### CODE REVIEW

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Areas/Identity/Pages/Account/ExternalLogin.cshtml.cs [hunk 1]
```diff
@@ -68,15 +68,15 @@ public IActionResult OnPost(string provider, string returnUrl = null)
             return new ChallengeResult(provider, properties);
         }
 
-        public async Task<IActionResult> OnGetCallbackAsync(string providedReturnUrl = null, string remoteError = null)
+        public async Task<IActionResult> OnGetCallbackAsync(string returnUrl = null, string remoteError = null)
         {
-            var returnUrl = providedReturnUrl ?? Url.Content("~/");
+            var returnUrlOrFallback = returnUrl ?? Url.Content("~/");
```
Comment (ntsim) [inline src/GovUk.Education.ExploreEducationStatistics.Admin/Areas/Identity/Pages/Account/ExternalLogin.cshtml.cs:73]: Not sure how this fixes the problem. It looks like we've just renamed some variables?

Comment (markhiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Admin/Areas/Identity/Pages/Account/ExternalLogin.cshtml.cs:73]: It's the renaming of the argument name to `returnUrl` that fixed it. Duncan noticed that newer version of IdentityServer, the `OnGetCallbackAsync` method's argument name wasn't `providedReturnUrl`, so we speculatively changed it to `returnUrl` and then redirection worked. Dunc suggested IdentityServer uses reflection or something.

Comment (ntsim) [reply src/GovUk.Education.ExploreEducationStatistics.Admin/Areas/Identity/Pages/Account/ExternalLogin.cshtml.cs:73]: Wow that's just mad. Can you leave a comment on the method saying that the parameter names are important?

Comment (markhiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Admin/Areas/Identity/Pages/Account/ExternalLogin.cshtml.cs:74]: Done. I've also updated the PR description.

**Review by ntsim — APPROVED** [review]

**Pull request merged**